### PR TITLE
[NFC] Remove uses of supportsTypedPointers()

### PR DESCRIPTION
**Pull request by MrSidims**
Also fix 'unused variable' warning.

### CODE REVIEW

**Review by svenvh — APPROVED** [review]

File: lib/SPIRV/SPIRVWriter.cpp [hunk 1]
```diff
@@ -1804,7 +1804,7 @@ LLVMToSPIRVBase::transValueWithoutDecoration(Value *V, SPIRVBasicBlock *BB,
       auto BV = transConstant(Init);
       assert(BV);
       return mapValue(V, BV);
-    } else if (ConstantExpr *ConstUE = dyn_cast_or_null<ConstantExpr>(Init)) {
+    } else if (Init && isa<ConstantExpr>(Init)) {
```
Comment (asudarsa) [inline lib/SPIRV/SPIRVWriter.cpp:1807]: Just curious...why is this change part of this PR? Thanks

File: lib/SPIRV/SPIRVWriter.cpp [hunk 2]
```diff
@@ -2766,10 +2766,6 @@ SPIRVValue *LLVMToSPIRVBase::oclTransSpvcCastSampler(CallInst *CI,
                                                      SPIRVBasicBlock *BB) {
   assert(CI->getCalledFunction() && "Unexpected indirect call");
   llvm::Function *F = CI->getCalledFunction();
-  auto FT = F->getFunctionType();
```
Comment (asudarsa) [inline lib/SPIRV/SPIRVWriter.cpp:2769]: Same concern as above. Why is this change par of this PR? Is this PR accomplishing more that what the description says?

Thanks

**Review by asudarsa — APPROVED** [review]
LGTM. THANKS

**Comment by asudarsa** [comment]
Hi @MrSidims 

Please address CI issues.

Thanks

**Comment by MrSidims** [comment]
Restarting CI

**Pull request merged**